### PR TITLE
Colw/signup steps

### LINKED ISSUE #2894: let's use steps for session screens
**Is your feature request related to a problem? Please describe.**

- the session forms are long and condensed. especially on mobile and in the browser extension. ideally, the user would not scroll during onboarding.

**Describe the solution you'd like**

- let's add a steps component like in the action modal for creating an account and recovering from backup. each step will include a sentence or two explaining what's happening.

**Create**
step 1 - account name (and email)
step 2 - password and password confirmation
step 3 - backup code with checkbox
step 4 - positive reinforcement / congrats! 

**Recover**
step 1 - backup code
step 2 - confirm address
step 3 - account name (and email)
step 4 - password and password confirmation
step 5 - positive reinforcement / congrats

### PR DESCRIPTION
**Pull request by colw**
Closes #2894 

**Description:**

- Use separate module to keep signup form
- Break signup into separate steps

Up for comment before cleaning code.

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by jbibla** [comment]
looks good. few thoughts: 

- perhaps we could name them by their content as opposed to 1,2,3
- should we enable this stuff in the browser with a warning @faboweb ?
- this should apply to recover as well as sign up

**Comment by colw** [comment]
> looks good. few thoughts:
> 
> * perhaps we could name them by their content as opposed to 1,2,3
> * should we enable this stuff in the browser with a warning @faboweb ?
> * this should apply to recover as well as sign up

- I've renamed them now, and cleaned the code.
- FYI, I've enabled the warning just for the build previews. But maybe we could keep it?
- I'll do that once we're happy with this.

**Comment by faboweb** [comment]
> should we enable this stuff in the browser with a warning @faboweb ?

Happy to enable in the browser. But let's do in another PR. #2992 2992

File: src/components/common/InsecureModeWarning.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<template>
+  <div>
+    <p>
+      Creating an address in the browser is unsafe. To offer you a secure
```
Comment (faboweb) [inline src/components/common/InsecureModeWarning.vue]: Proposing a new warning (branding + extension is available)

```
Did you know that storing your key in the browser is considered unsafe? We offer a more secure way to storing your keys via our _browser extension_. Check it out in a quiet minute.
```

Comment (colw) [reply src/components/common/InsecureModeWarning.vue]: I like that

**Comment by mariopino** [comment]
Both `/create` and `/recover` flows working! Lot of testing work pending...

**Comment by jbibla** [comment]
awesome work here @colw and @mariopino 

few thoughts about the flow / ux:
- the buttons should say "next" until the final step "create"
- maybe the warning could be displayed _only_ once, as the first step (in the browser)

> A message should go here.

made me laugh 😆

**Comment by mariopino** [comment]
> awesome work here @colw and @mariopino
> 
> few thoughts about the flow / ux:
> 
> * the buttons should say "next" until the final step "create"
> * maybe the warning could be displayed _only_ once, as the first step (in the browser)
> 
> > A message should go here.
> 
> made me laugh 😆

I changed the button messages in both flows and display the warning in the first step of create.

Would be nice some inspiration for the success pages btw :-)

**Comment by jbibla** [comment]
few thoughts: 

- unless the session form is longer than the screen, it should not scroll
- buttons should only be full width on mobile, tablet and up can be regular sized
- danger screen can be on its own as first step (only in browser)
- recover screen has some obvious styling issues on mobile
- success screen button should say "Go to Lunie" instead of "Back to Lunie"

- have you looked at the impact of these changes on the extension?

**Comment by jbibla** [comment]
another thought: 

- let's make the back button a bit smaller and put it in a little circle like this:
Screen Shot 2019-10-03 at 12 55 41 PM

**Comment by mariopino** [comment]
Working in progress:

- Buttons are now small and right aligned by default in all steps, and full width in screens < 667px wide.
- Success screen button now says "Go to Lunie" instead of "Back to Lunie"
- Now back button looks like this:

image

File: src/vuex/modules/session.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ export default () => {
   const state = {
     developmentMode: config.development, // can't be set in browser
     experimentalMode: config.development, // development mode, can be set from browser
-    insecureMode: false, // show the local signer
+    insecureMode: true, // show the local signer
```
Comment (faboweb) [inline src/vuex/modules/session.js]: don't forget to remove this again ^^

Comment (faboweb) [reply src/vuex/modules/session.js]: dangerous to set it in code. better use the query parameter: `?insecure=true`

Comment (mariopino) [reply src/vuex/modules/session.js]: > don't forget to remove this again ^^

Yeah, Colin was farsighted and wrote a test to check that it's false :-)

Comment (faboweb) [reply src/vuex/modules/session.js]: Good guy Colin

File: changes/colw_signup-steps [hunk 3]
```diff
@@ -0,0 +1 @@
+[Changed] [#2894](https://github.com/cosmos/lunie/issues/2894) Refactor signin steps into separate screens, including a success screen. @colw
```
Comment (faboweb) [inline changes/colw_signup-steps]:
```suggestion
[Changed] [#2894](https://github.com/cosmos/lunie/issues/2894) Refactor signin steps into separate screens, including a success screen. @colw @mariopino
```

File: src/components/common/DangerZoneWarning.vue [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<template>
```
Comment (faboweb) [inline src/components/common/DangerZoneWarning.vue]: Nice extraction

File: src/components/common/InsecureModeWarning.vue (hunk 1 above)
Comment (faboweb) [reply src/components/common/InsecureModeWarning.vue]: @mariopino did you see this comment?

File: src/initializeApp.js [hunk 5]
```diff
@@ -18,8 +18,11 @@ function setOptions(urlParams, store) {
   if (urlParams.rpc) {
     store.commit(`setRpcUrl`, urlParams.rpc)
   }
-  if (config.mobileApp || urlParams.insecure) {
-    store.commit(`setInsecureMode`)
+  if (config.mobileApp || urlParams.insecure === `true`) {
+    store.commit(`setInsecureMode`, true)
+  }
+  if (urlParams.insecure === `false`) {
```
Comment (faboweb) [inline src/initializeApp.js]: This is the default, why set it?

File: src/routes.js [hunk 6]
```diff
@@ -127,6 +127,57 @@ export default [
       feature: "Session"
     }
   },
+  {
```
Comment (faboweb) [inline src/routes.js]: The whole routing schema looks excessive. :/ But let's leave it for now.

File: src/components/common/InsecureModeWarning.vue (hunk 1 above)
Comment (mariopino) [reply src/components/common/InsecureModeWarning.vue]: > @mariopino did you see this comment?

Yes but for some reason i thought it was already done.

What link should we include in ` _browser extension_` link? Chrome store?

Comment (mariopino) [reply src/components/common/InsecureModeWarning.vue]: > > @mariopino did you see this comment?
> 
> Yes but for some reason i thought it was already done.
> 
> @faboweb What link should we include in ` _browser extension_` link? Chrome store?

Autoanswer: http://bit.ly/lunie-ext fits well here

File: src/initializeApp.js (hunk 5 above)
Comment (mariopino) [reply src/initializeApp.js]: The flag can be set to true in store and you should also have the possibility to change to false via url param. I'm happy to revert it if you find it's not useful.

File: src/routes.js (hunk 6 above)
Comment (mariopino) [reply src/routes.js]: Agree, needed by current approach, but we can get rid off at refactor.

File: src/components/common/InsecureModeWarning.vue (hunk 1 above)
Comment (mariopino) [reply src/components/common/InsecureModeWarning.vue]: image

**Comment by faboweb** [comment]
Typo: "out" should be "our"

Mario Pino <notifications@github.com> schrieb am Di., 15. Okt. 2019, 12:50:

**Comment by faboweb** [comment]
In which usecase would you set to insecure=false via the url?

Mario Pino <notifications@github.com> schrieb am Di., 15. Okt. 2019, 12:39:

File: src/components/common/DangerZoneWarning.vue [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  <div class="danger-zone">
+    <h2>DANGER ZONE</h2>
+    <p>
+      Did you know that storing your key in the browser is considered unsafe? We
```
Comment (jbibla) [inline src/components/common/DangerZoneWarning.vue]:
```suggestion
      Did you know that creating your address in the browser is considered unsafe? We
```

File: src/components/common/DangerZoneWarning.vue [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      offer a more secure way to storing your keys via our
```
Comment (jbibla) [inline src/components/common/DangerZoneWarning.vue]:
```suggestion
      offer a more secure way to do this with our
```

File: src/components/common/DangerZoneWarning.vue [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      <a
+        href="http://bit.ly/lunie-ext"
+        target="_blank"
+        rel="noopener norefferer"
+        >browser extension</a
+      >. Check it out in a quiet minute. Soon you will also have out mobile app
```
Comment (jbibla) [inline src/components/common/DangerZoneWarning.vue]:
```suggestion
      >. Check it out in a quiet minute.
```

File: src/components/common/DangerZoneWarning.vue [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+      available to store your keys. Stay tuned!
```
Comment (jbibla) [inline src/components/common/DangerZoneWarning.vue]:
```suggestion

```

File: src/components/common/TmSessionSignUp.vue [hunk 11]
```diff
@@ -1,249 +1,98 @@
...
-          <TmFormMsg
-            v-if="
-              $v.fields.signUpWarning.$error &&
-                !$v.fields.signUpWarning.required
-            "
-            name="Recovery confirmation"
-            type="required"
-          />
-        </TmFormGroup>
+          <DangerZoneWarning />
```
Comment (jbibla) [inline src/components/common/TmSessionSignUp.vue]: can we put this above line 4?

File: src/components/common/TmSessionSignUpSeed.vue [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+      <div v-if="session.insecureMode">
+        <div class="session-main">
+          <Steps
+            :steps="[`Create`, `Password`, `Seed`, `Success`]"
+            active-step="Seed"
+          />
+          <TmFormGroup
+            field-id="sign-up-seed"
+            class="sign-up-seed-group"
+            field-label="Seed phrase"
```
Comment (jbibla) [inline src/components/common/TmSessionSignUpSeed.vue]:
```suggestion
            field-label="Backup Code"
```

File: src/components/common/TmSessionImportSuccess.vue [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+    </h2>
+    <div class="session-main">
+      <Steps
+        :steps="[`Recover`, `Name`, `Password`, `Success`]"
+        active-step="Success"
+      />
+      <p>Your account has been created</p>
+    </div>
+    <div class="session-footer">
+      <TmBtn value="Back to Lunie" @click.native="$router.push('/')" />
```
Comment (jbibla) [inline src/components/common/TmSessionImportSuccess.vue]:
```suggestion
      <TmBtn value="Go to Lunie" @click.native="$router.push('/')" />
```

**Comment by faboweb** [comment]
image

We should unify the wording. It says seed in the steps and in the confirmation checkbox. @jbibla preferences?

**Comment by faboweb** [comment]
The check for the duplicate name is done too late in the process. It should be done immediately on the name step.

**Comment by faboweb** [comment]
I think the step points should move a bit more apart. They are very squished together:
image

File: src/components/common/InsecureModeWarning.vue (hunk 1 above)
Comment (faboweb) [reply src/components/common/InsecureModeWarning.vue]: image

**Comment by jbibla** [comment]
> I think the step points should move a bit more apart. They are very squished together:

what if we remove step 4 altogether?

**Comment by mariopino** [comment]
> The check for the duplicate name is done too late in the process. It should be done immediately on the name step.

Done!

**Pull request merged**